consensus/src/simplex/actors/voter/actor.rs

```diff
@@ -1834,6 +1894,35 @@ impl<
                         continue;
                     }
                 },
+                (context, certified) = certify_wait => {
+                    // Clear certify waiter
+                    pending_certify = None;
+
+                    // Return early if the proposal is not certified
+                    match certified {
+                        Ok(false) => {
+                            debug!(round = ?context.round, "proposal is not certified");
+                            continue;
+                        }
+                        Err(err) => {
+                            debug!(?err, round = ?context.round, "failed to certify proposal");
+                            continue;
+                        }
+                        _ => {},
+                    };
+
+                    // Ensure downstream notify runs for this view (finalize/finalization attempts)
+                    view = context.view();
+                    let round = self.views.get_mut(&view).unwrap();
+                    round.certified_proposal = true;
+
+                    // If we already have a notarization for this view, advance now.
+                    let notarization = round.notarization.as_ref().expect("certified proposal should have notarization");
+                    let seed = self
+                        .scheme
+                        .seed(notarization.round(), &notarization.certificate);
+                    self.enter_view(view + 1, seed);
```

Comment:
this will be handled at the end of the loop already (or at least it should be if that's been refactored out IMHO)?

Reply:
I think we could abstract this logic for consistency with `handle_certification`